PerfNext/public/lib/js/util.js

```diff
@@ -246,6 +248,127 @@ function getEspressoPlatform() {
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'Windows';
+    	break;
+    case 'AIX PPC 64 bit':
+    	cpuInfo['cpuType'] = 'PPC';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'AIX';
+    	break;
+    case 'Linux Intel x86 32 bit':
+    	cpuInfo['cpuType'] = 'x86';
+    	cpuInfo['bits'] = '32';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux AMD x64 64 bit':
+    	cpuInfo['cpuType'] = 'AMD64';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux PPC 32 bit':
+    	cpuInfo['cpuType'] = 'PPC';
+    	cpuInfo['bits'] = '32';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux PPC 64 bit':
+    	cpuInfo['cpuType'] = 'PPC';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux PPCLE 64 bit':
+    	cpuInfo['cpuType'] = 'PPCLE';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux z Series 31 bit':
+    	cpuInfo['cpuType'] = 'z Series';
+    	cpuInfo['bits'] = '31';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'Linux z Series 64 bit':
+    	cpuInfo['cpuType'] = 'z Series';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'Linux';
+    	break;
+    case 'MVS z Series 31 bit':
+    	cpuInfo['cpuType'] = 'z Series';
+    	cpuInfo['bits'] = '31';
+    	cpuInfo['OS'] = 'MVS';
+    	break;
+    case 'MVS z Series 64 bit':
+    	cpuInfo['cpuType'] = 'z Series';
+    	cpuInfo['bits'] = '64';
+    	cpuInfo['OS'] = 'MVS';
+    	break;
+	default:
+		console.log('getCPUinfo() ERROR OS: '+OS);	
+    }
+    return cpuInfo;
+}
+
+function getAllMatchingMachines(cpuInfo) {
+	console.log("Entering getAllMatchingMachines(cpuInfo)")
+	var allHosts = allmachinesInfo.getElementsByTagName("hosts")[0].childNodes.values();
+ 	var validMachines = {};
+ 	
+	for(const host of allHosts) {
+		if (host.id != null) {
+			var hostNode = host;
+			
+			var cpuMatch = false;
+	 		var bitsMatch = false;
+	 		var OSMatch = hostNode.getElementsByTagName('os')[0].getAttribute('type') == cpuInfo['OS'];
+			var jobGen = hostNode.getElementsByTagName('jobgenerator')[0].getAttribute('enabled') == "true";
+			
+			// Early exit if OS or jobGen is false
+			if (!OSMatch || !jobGen) {
+				continue;
+			}
+			var machineId = hostNode.getElementsByTagName('machine')[0].getAttribute('id');
+			var hostName = hostNode.getElementsByTagName('name')[0].innerHTML;
+			var machineNode = allmachinesInfo.querySelector('[id="' + machineId + '"]');
+			
+			var cpuTypeArray  = machineNode.getElementsByTagName('cputype');
+			for (const cpuType of cpuTypeArray) {
+				if (cpuType.innerHTML == cpuInfo['cpuType']) {
+					cpuMatch = true;
+					break;
+				}
+			}
+			
+			var bitsArray  = machineNode.getElementsByTagName('bits');
+			for (const bits of bitsArray) {
+				if (bits.innerHTML == cpuInfo['bits']) {
+					bitsMatch = true;
+					break;
+				}
+			}
+			if (cpuMatch && bitsMatch) {
+				validMachines[hostName] = hostNode.id;
```

Comment:
Please try using dot notation instead of bracket notation in all places wherever possible since dot notation is faster to write and clearer to read.

Reply:
Can't use dot notation with variables in this case hostName, other wise the array would just have a single entry with key hostName.

Reply:
Yeah that was more of a general comment to use dot notation "wherever possible". Don't remember whether I saw bracket notation being used somewhere here. Anyway this looks good!